mobile-widgets/qml/DiveDetails.qml

```diff
@@ -262,7 +262,7 @@ Kirigami.Page {
 			boundsBehavior: Flickable.StopAtBounds
 			maximumFlickVelocity: parent.width * 5
 			orientation: ListView.Horizontal
-			highlightFollowsCurrentItem: true
+			highlightFollowsCurrentItem: false
```

Comment:
what worries me is that I changed this in https://github.com/Subsurface-divelog/subsurface/commit/783f9ee565adc1 specifically to FIX the issue that you are now reporting. We have changed this property a couple of times during the development of Subsurface-mobile...

Reply:
I see the history a little different. The commit 783f9ee first introduces the highlightFollowsCurrentItem to set it to (according to the current Qt docs) the default value. I looked back to end 2016 trough all commits touching this file. So, in hindsight, this looks like an irrelevant line.

Further, there is commit 59417342c7, that is supposed to deal with the enumerate problem. Not touching this property.

Also, we are talking (for both mentioned commits) over Kirigami 1 (Qt version I'm not sure, might be 5.9.1 or 5.9.0).

Finally, the behavior I now see (on 5.9.2) is different from the problem I used to call "the infamous enumeration problem". That old problem was repeated selecting a dive from the dive list, and suddenly the app froze, and (from logging) we could see that it was rendering all the dives in the list, eventually running out of memory (on some phones, depending of the size of the logbook).

The issue fixed here is: select a dive , back to the dive list, select another dive, and the app starts scrolling (on screen, like a slideshow) to the new dive. Fully defined, and seemingly intentional behavior. That we encounter it now, is most likely due to some bug fixing on Qt side).

And I fully share your worries, this all seems a little fragile.